hni-admin-service/src/main/java/org/hni/admin/service/UserReportsController.java

```diff
@@ -33,34 +36,92 @@
 @Api(value = "/reports/view", description = "Reports for all user type")
 @Component
 @Path("/reports/view")
-public class UserReportsController {
-	
+public class UserReportsController extends AbstractBaseController {
+
 	private static final Logger _LOGGER = LoggerFactory.getLogger(UserOnboardingController.class);
 
 	@Inject
 	private UserReportService userReportService;
+
 	@GET
 	@Path("/NGO")
 	@Produces({ MediaType.APPLICATION_JSON })
 	@ApiOperation(value = "Service for getting basic detail of NGO", notes = "", response = List.class, responseContainer = "")
 	public Response getAllNgo() {
-		
-		Map<String,Object> response = new HashMap<>();
-		try
-		{
-		List<NgoBasicDto> ngo =  userReportService.getAllNgo();
-		response.put("headers", HNIUtils.getHeader(Constants.USER_TYPES.get("ngo")));
-		response.put("data", ngo);
+
+		Map<String, Object> response = new HashMap<>();
+		try {
+			List<NgoBasicDto> ngo = userReportService.getAllNgo();
+			response.put("header", HNIUtils.getHeader(Constants.USER_TYPES.get("ngo")));
+			response.put("data", ngo);
+		} catch (Exception e) {
+			_LOGGER.error("Error in get Ngo Service:" + e.getMessage(), e);
+			response.put(Constants.RESPONSE, Constants.ERROR);
 		}
-		catch(Exception e){
-			_LOGGER.error("Error in get Ngo Service:"+e.getMessage(), e);
+		response.put(Constants.RESPONSE, Constants.SUCCESS);
+		return Response.ok(response).build();
+
+	}
+
+	@GET
+	@Path("/customersbyrole")

```

Comment:
use proper naming conventions when web service resource path defined. Nouns should be separated by "/"